client/src/components/share-workspace-dialog.tsx

```diff
@@ -16,30 +15,96 @@ import {
   SelectTrigger,
   SelectValue,
 } from "@/components/ui/select";
+import { useEffect, useState } from "react";
+
+const ShareWorkspaceDialog = ({ workspaceId }: { workspaceId: string }) => {
+  const [open, setOpen] = useState(false);
+  const [users, setUsers] = useState([]);
+  const [selectedUser, setSelectedUser] = useState(null);
+
+  useEffect(() => {
+    const jwt = localStorage.getItem("jwt");
+    const requestOptions = {
+      method: "GET",
+      headers: {
+        Authorization: `Bearer ${jwt}`,
+        "Content-Type": "application/json",
+      },
+    };
+
+    fetch("http://localhost:8000/api/users", requestOptions)
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error("Network response was not ok");
+        }
+
+        return response.json();
+      })
+      .then((data) => {
+        setUsers(data);
```

Comment:
Should we add a check to make sure we don't include the currently logged in user themself? If so, we can either filter out the user before we set it here, or edit the API call to only fetch users that aren't the current user.